File: apps/registry/src/modules/core/application/use-cases/exemplo.ts
import { Inject, Injectable } from '@nestjs/common';
import { CLOCK, ID_GENERATOR, type Clock, type IdGenerator } from '../ports.js';
import type { ExemploCommand, ExemploResult } from '../dto.js';

/**
 * Caso de uso de exemplo. Troque por um real e apague este.
 *
 * Regras do template (doc 03, secao 3.2):
 *   - recebe um comando, nunca o `Request` do Express;
 *   - fala so com ports;
 *   - todo caminho de erro tem teste.
 */
@Injectable()
export class Exemplo {
  constructor(
    @Inject(CLOCK) private readonly clock: Clock,
    @Inject(ID_GENERATOR) private readonly ids: IdGenerator,
  ) {}

  async execute(command: ExemploCommand): Promise<ExemploResult> {
    void command;
    void this.clock;
    return Promise.resolve({ id: this.ids.next() });
  }
}

File: apps/evaluation/src/evaluation/application/use_cases/exemplo.py
"""Caso de uso de exemplo. Troque por um real e apague este."""

from __future__ import annotations

import uuid
from dataclasses import dataclass

from evaluation.application.dto import ExemploCommand
from evaluation.application.ports import ExemploRepository
from evaluation.domain.entities import Exemplo


@dataclass(slots=True)
class ExecutarExemplo:
    repository: ExemploRepository

    async def execute(self, command: ExemploCommand) -> Exemplo:
        exemplo = Exemplo(id=str(uuid.uuid4()), project_id=command.project_id)
        await self.repository.save(exemplo)
        return exemplo
